src/main.rs

```diff
@@ -1,3 +1,829 @@
+    #[clap(flatten)]
+    manifest: clap_cargo::Manifest,
+    #[clap(flatten)]
+    workspace: clap_cargo::Workspace,
+    #[clap(flatten)]
+    features: clap_cargo::Features,
+
+    // Top-level flags
+    /// Do not pull in new "audits".
+    #[clap(long)]
+    locked: bool,
+
+    /// How verbose logging should be (log level).
+    #[clap(short, long, arg_enum)]
+    #[clap(default_value_t = Verbose::Warn)]
+    verbose: Verbose,
+
+    /// Instead of stdout, write output to this file.
+    #[clap(long)]
+    output_file: Option<PathBuf>,
+
+    /// Instead of stderr, write logs to this file (only used after successful CLI parsing).
+    #[clap(long)]
+    log_file: Option<PathBuf>,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// initialize cargo-vet for your project
+    Init(InitArgs),
+    /// Fetch the source of `$crate $version`
+    Fetch(FetchArgs),
+    /// Yield a diff against the last reviewed version.
+    Diff(DiffArgs),
+    /// Mark `$crate $version` as reviewed with `$message`
+    Certify(CertifyArgs),
+    /// Mark `$crate $version` as unacceptable with `$message`
+    Forbid(ForbidArgs),
+    /// Suggest some low-hanging fruit to review
+    Suggest(SuggestArgs),
+    /// ??? List audits mechanisms ???
+    Audits(AuditsArgs),
+    /// Print --help as markdown (for generating docs)
+    HelpMarkdown(HelpMarkdownArgs),
+}
+
+#[derive(clap::Args)]
+struct InitArgs {}
+
+#[derive(clap::Args)]
+struct FetchArgs {
+    krate: String,
+    version: String,
+}
+
+#[derive(clap::Args)]
+struct DiffArgs {}
+
+#[derive(clap::Args)]
+struct CertifyArgs {
+    krate: String,
+    version: String,
+    message: String,
+}
+
+#[derive(clap::Args)]
+struct ForbidArgs {
+    krate: String,
+    version: String,
+    message: String,
+}
+
+#[derive(clap::Args)]
+struct SuggestArgs {}
+
+#[derive(clap::Args)]
+struct AuditsArgs {}
+
+#[derive(clap::Args)]
+struct HelpMarkdownArgs {}
+
+/// Logging verbosity levels
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ArgEnum)]
+enum Verbose {
+    Off,
+    Error,
+    Warn,
+    Info,
+    Debug,
+    Trace,
+}
+
+#[derive(serde::Deserialize)]
+struct Config {
+    // Reserved for future use, if not present version=1 assumed.
+    // (not sure whether this versions the format, or semantics, or...
+    // for now assuming this species global semantics of some kind.
+    version: Option<u64>,
+    store: Option<Store>,
+    audits: Option<Vec<String>>,
+}
+#[derive(serde::Deserialize)]
+struct Store {
+    path: Option<PathBuf>,
+}
+
+// FIXME: Probably want this to be a tree, and for queries to
+// be keyed off the "current package" but unclear what that would *mean*
+// so for now assume there is a linear chain of overrides. Possible
+// that any "tree-like" situation wants to be an error...
+
+/// All available configuration files, overlaying eachother.
+/// Generally contains: `[Default, Workspace, Package]`
+struct Configs(Vec<Config>);
```

Comment:
I currently have this setup as a cascade of the 3 sources of config, but in principle package tomls being able to have configs implies a tree-like structure and I have no idea *what* that would mean for cargo-vet's execution, which afaict is currently conceptualized to operate globally and not per-package.

Reply:
Ah, this is partially resolved by making the config.toml file! Still a bit unclear what it would mean for a crate to "override" the workspace for the `store.path` though.

Reply:
Yeah conceptually it's global for the workspace or project. Individual crates shouldn't be able to override it.

Reply:
Should we hard error, or just warn (and discard the package's values) if we find both?